Assets/BossRoom/Scripts/Shared/Game/Entity/NetworkCharacterState.cs

```diff
@@ -113,21 +114,32 @@ public LifeState LifeState
         public override void OnNetworkSpawn()
         {
             if (!IsServer) return;
-            HitPoints = GetInitialHitPoints();
+            SetInitialHitPoints();
         }
 
-        int GetInitialHitPoints()
+        void SetInitialHitPoints()
         {
+            HitPoints = CharacterClass.BaseHP.Value;
+
             if (!IsNpc)
             {
                 SessionPlayerData? sessionPlayerData = SessionManager<SessionPlayerData>.Instance.GetPlayerData(OwnerClientId);
                 if (sessionPlayerData is { HasCharacterSpawned: true })
                 {
-                    return sessionPlayerData.Value.CurrentHitPoints;
+                    StartCoroutine(SetReconnectingHitPointsCoroutine(sessionPlayerData.Value.CurrentHitPoints));
                 }
             }
+        }
 
-            return CharacterClass.BaseHP.Value;
+        IEnumerator SetReconnectingHitPointsCoroutine(int hitPoints)
+        {
+            // Wait until end of frame to make sure other NetworkBehaviors for this character had time to spawn.
+            yield return new WaitForEndOfFrame();
```

Comment:
Is this necessary? If other NetworkBehaviours were subscribed to the OnValueChanged callback, that should fire regardless, right? What other considerations are there?

Reply:
From my tests, it doesn't fire. ServerCharacter and ServerAnimationHandler both register to LifeState's OnValueChanged inside their OnNetworkSpawned, but if I change its value in NetworkCharacterState's OnNetworkSpawn, it is not caught by them. Concretely, this means the animation trigger for fainting is not sent and the reconnecting character is simply immobile, but still standing and doing the anticipation animations when trying to move

Reply:
Hold on. After rethinking it, this should actually be done inside `ServerAnimationHandler`. You can call `OnLifeStateChanged` directly as you're subscribing to the `OnValueChanged` callback for `LifeState`. It's a common pattern in the codebase for initializing based on a `NetworkVariable`'svalue on `OnNetworkSpawn()`.